modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -2787,7 +2787,7 @@ struct LinearScratchDesc {
                     outW * sizeof(index_t)     +
                     outH * sizeof(alpha_t)     +
                     outH * sizeof(index_t) * 2 +
-                     inW * sizeof(T) * lpi * chanNum;
+                    (inW + 1) * sizeof(T) * lpi * chanNum;
```

Comment:
Its only hides the problem, but this doesn't resolve that actually.

Out of bound access should not be resolved through increasing of buffers size.
Loop constraints should be properly checked instead.

Reply:
It is because we load with intrinsic `_mm_lddqu_si128` 16 bytes of memory. With this 16 bytes of memory we load 3 packs (1 pack for 1 channel) of 4 pixels ([r1,r2,r3,r4],[g1,g2,g3,g4],[b1,b2,b3,b4]). Each pixel in the pack is a pixel from different rows. But also we load 1 additional pack for first channel of following 4 pixels from the same rows - ([r5,r6,r7,r8]).
If we map for output pixel - end pixel on the right side from input image -> we can retrieve only 3 channels for 4 rows, but not additional 1st channel from non-existent neighboring pixel. So, the solution was to provide this neighboring channel.

Reply:
Do you propose to add check for crossing the boundary and form the 128bit vector another way?

Reply:
All SIMD loops must have tail processing which is properly handled.
Currently processing functions still have no checks and no correct constraints.

Also we may have SIMD256 in the future, so 32 bytes will be read at once. And code would go to read out of buffer again and again.

---

Reading 16 bytes and using just 12 bytes is a strong signal of useless work in non-efficient code.
This is why we read 48 bytes (16*3) in case of SIMD128 and RGB in other OpenCV code. Look for `v_load_deinterleave()` calls.
As I wrote before in original PR, instead of writing a completely new code please take a look on already existed code (we don't want to reinvent the bicycles with square wheels). This question is not about the result, it is about the development process. /cc @dmatveev

Reply:
@alalek thanks for pointing this out - actually the last PR on Resize was supposed to align it with the OpenCV way of reading pixels. We'll check this out